otel-tests/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+use opentelemetry::sdk::trace;
+use opentelemetry::sdk::Resource;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::WithExportConfig;
+use std::sync::Once;
+use tracing_subscriber::filter::LevelFilter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+use tracing_subscriber::Layer;
+use tracing_subscriber::Registry;
+
+pub use futures;
+pub use opentelemetry;
+pub use otel_tests_macro::otel_test;
+pub use tokio;
```

Comment:
❓ Might it be preferable to move these re-exported dependencies to the `otel-tests-macro` crate instead of re-exporting them? From the perspective of any crate that depends on `otel-tests` it doesn't make sense that it should re-export anything other than the `otel_test` macro.

Reply:
IIRC depending on a crate transitively doesn't bring it into scope?

Reply:
It's definitely not a big deal. But I think what I'm expecting is `rust-analyzer` to, for example, offer me an `otel-tests::futures::Future` import when I try to auto-import `Future`. Just a minor annoyance if I'm understanding how this works.

Reply:
That's true. On the other hand, I don't want to force transitive dependencies to declare a whole lot of extra hidden dependencies.

Will RA still show up for `#[doc(hidden)]` stuff? Maybe we could just `#[doc(hidden)]` all of them...

Reply:
Haha, apparently we can hide re-exports that way because of this bug: https://github.com/rust-lang/rust/issues/59368.

But would that also break the macro?

Reply:
I think this just affects rustdoc, but I think RA might ignore doc hidden stuff too

Reply:
It does say

> Both of the re-exports above work just fine

in the issue I linked. I should read what I link to lol.

>  but I think RA might ignore doc hidden stuff too

Interesting! Can't hurt to try.

Reply:
That didn't seem to work for intellij - but it doesn't come first at least
![image](https://user-images.githubusercontent.com/6688948/179222657-18d20339-f6e9-42fc-b614-cf8c64e4d867.png)

Reply:
> That didn't seem to work for intellij - but it doesn't come first at least
> ![image](https://user-images.githubusercontent.com/6688948/179222657-18d20339-f6e9-42fc-b614-cf8c64e4d867.png)

I think this is good enough!